### ISSUE TITLE
processingInstructions crop and cropVariant will be ignored

### ISSUE THREAD

**Issue by mediaessenz** [bithost-gmbh/pdfviewhelpers#136]
I seems, that the process instruction properties "crop" and "cropVariant" will be ignored completely.

I try to do something like this:
```
<f:variable name="records" value="{v:content.get(column:0, pageUid:123, limit:1, slide:'-1')}" />
<f:for each="{records}" as="record">
    <v:content.resources.fal field="image" as="images" record="{record}">
        <f:for each="{images}" as="image">
            <f:if condition="{image}">
                <pdf:image src="{image.id}" processingInstructions="{crop: image.crop, cropVariant:'default'}" width="100"/>
            </f:if>
        </f:for>
    </v:content.resources.fal>
</f:for>
```
Independent of any cropping inside the added images, they get not cropped.
If I add this debug output to the upper code right after the pdf:image:
```
<pdf:html>
    <pre>{image.crop -> f:debug(inline:1, plainText: 1)}</pre>
</pdf:html>
```
I get the correct settings array, which looks like this:
```
{
   "default":{
      "cropArea":{
         "height":1,
         "width":0.441,
         "x":0,
         "y":0
      },
      "selectedRatio":"NaN",
      "focusArea":{
         "x":0,
         "y":0,
         "width":0,
         "height":0
      }
   },
   "large":{
      "cropArea":{
         "x":0,
         "y":0,
         "width":1,
         "height":1
      },
      "selectedRatio":"NaN",
      "focusArea":{
         "x":0,
         "y":0,
         "width":0,
         "height":0
      }
   },
   "medium":{
      "cropArea":{
         "x":0,
         "y":0,
         "width":1,
         "height":1
      },
      "selectedRatio":"NaN",
      "focusArea":{
         "x":0,
         "y":0,
         "width":0,
         "height":0
      }
   },
   "small":{
      "cropArea":{
         "x":0,
         "y":0,
         "width":1,
         "height":1
      },
      "selectedRatio":"NaN",
      "focusArea":{
         "x":0,
         "y":0,
         "width":0,
         "height":0
      }
   },
   "extrasmall":{
      "cropArea":{
         "x":0,
         "y":0,
         "width":1,
         "height":1
      },
      "selectedRatio":"NaN",
      "focusArea":{
         "x":0,
         "y":0,
         "width":0,
         "height":0
      }
   }
}
```

**Comment by mediaessenz** [comment]
After doing some research, I think the problem is the applyProcessingInstructions call inside the pdf:image viewhelper.
I think the overhanded processing instructions especial crop and cropVariant have to be transformed before they will work.
The call of this method ends in the process method of the LocalCropScaleMaskHelper class, which does somethink like this if a crop configuration is found:
```
$cropData = json_decode($configuration['crop']);
            if ($cropData) {
                $crop = implode(',', [(int)$cropData->x, (int)$cropData->y, (int)$cropData->width, (int)$cropData->height]);
            } else {
                $crop = $configuration['crop'];
            }

            list($offsetLeft, $offsetTop, $newWidth, $newHeight) = explode(',', $crop, 4);
```
This always results in "0,0,0,0", because the overgiven crop configuration is a multidimensional object with all crop variants, and not only the default one.
There is no cropVariant handling at all at this place.

Finally the type casting (int) in front of $cropData-> will convert all float values like "0.441" to 0

**Comment by maechler** [comment]
@mediaessenz You are absolutely right. I have to admit I never really tested crop variants and I thought that the image service will take care of it, but it does not.

A workaround would be to select the corresponding crop area yourself and pass it as `crop` attribute to the `processingInstructions`. The following could should show you a working example: 

```html
<f:variable name="crop" value="{default: {cropArea: {height:100,width:100,x:50,y:50}}}" />
<f:variable name="cropVariant" value="default" />
<f:variable name="cropArea" value="{crop.{cropVariant}.cropArea}" />
<f:variable name="cropForProcessingInstructions" value="{f:format.json(value: cropArea)}" />
<pdf:image processingInstructions="{crop: cropForProcessingInstructions}" src="EXT:pdfviewhelpers/Resources/Public/Examples/BasicUsage/Bithost.jpg" />
```

You probably just have to replace `crop` with `image.crop`. Does this work for you?

I will have to add a more user friendly support for crop variants to the `pdfviewhelpers` however.

**Comment by derhansen** [comment]
Just for the records (and not related to `processingInstructions`, but to cropVariants). When using the `pdf:html` ViewHelper, it is possible to use the plain Fluid Image ViewHelper to render the crop variant of an image.

```
<pdf:html>
    <f:image src="{model.image.uid}" treatIdAsReference="1" cropVariant="your-crop-variant" width="500"/>
</pdf:html>
```

**Comment by maechler** [comment]
I just released `v2.3.4` which fixes this issue.
